install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
+      shift 1
+      ;;
+
+    -p=* | --platform=*)
+      platform="${1#*=}"
+      shift 1
+      ;;
+    -b=* | --bin-dir=*)
+      BIN_DIR="${1#*=}"
+      shift 1
+      ;;
+    -a=* | --arch=*)
+      arch="${1#*=}"
+      shift 1
+      ;;
+    -B=* | --base-url=*)
+      base_url="${1#*=}"
+      shift 1
+      ;;
+    -V=* | --verbose=*)
+      verbose="${1#*=}"
+      shift 1
+      ;;
+    -f=* | -y=* | --force=* | --yes=*)
+      force="${1#*=}"
+      shift 1
+      ;;
+
+    *)
+      error "Unknown option: $1"
+      exit 1
+      ;;
   esac
 done
 
-if [ "${ARCH}" = "i386" ]; then
-  error "i386 builds are not yet available for Starship\n"
-  info "If you would like to see a build for your configuration,"
-  info "please create an issue requesting a build for ${MAGENTA}${ARCH}-${PLATFORM}${NO_COLOR}:"
-  info "${BOLD}${UNDERLINE}https://github.com/starship/starship/issues/new/${NO_COLOR}\n"
-  exit 1
-fi
+arch_check
+platform_check
+bin_dir_check
 
 printf "  %s\n" "${UNDERLINE}Configuration${NO_COLOR}"
 info "${BOLD}Bin directory${NO_COLOR}: ${GREEN}${BIN_DIR}${NO_COLOR}"
-info "${BOLD}Platform${NO_COLOR}:      ${GREEN}${PLATFORM}${NO_COLOR}"
-info "${BOLD}Arch${NO_COLOR}:          ${GREEN}${ARCH}${NO_COLOR}"
+info "${BOLD}Platform${NO_COLOR}:      ${GREEN}${platform}${NO_COLOR}"
+info "${BOLD}Arch${NO_COLOR}:          ${GREEN}${arch}${NO_COLOR}"
+printf "\n"
 
-# non-empty VERBOSE enables verbose untarring
-if [ -n "${VERBOSE-}" ]; then
-  VERBOSE=v
+# non-empty `verbose` enables verbose untarring
+if [ -n "$verbose" ]; then
   info "${BOLD}Verbose${NO_COLOR}: yes"
-else
-  VERBOSE=
+  printf "\n"
 fi
 
-echo
+tmp=$(mktmpdir "starship")
+case $platform in
+  pc-windows-msvc)
+    bin="$tmp/starship.exe"
+    ext="zip"
+    ;;
+  *)
+    bin="$tmp/starship"
+    ext="tar.gz"
+    ;;
+esac
+archive="$tmp/starship.$ext"
+
+base_url=${base_url:-"https://github.com/starship/starship/releases"}
+url="$base_url/latest/download/starship-$arch-$platform.$ext"
 
-EXT=tar.gz
-if [ "${PLATFORM}" = "pc-windows-msvc" ]; then
-  EXT=zip
+info "Tarball URL: ${UNDERLINE}${BLUE}${url}${NO_COLOR}"
+if [ -z "$force" ]; then
+  confirm "Install Starship ${GREEN}latest${NO_COLOR} to ${BOLD}${GREEN}${BIN_DIR}${NO_COLOR}?"
 fi
+path_check
 
-URL="${BASE_URL}/latest/download/starship-${ARCH}-${PLATFORM}.${EXT}"
-info "Tarball URL: ${UNDERLINE}${BLUE}${URL}${NO_COLOR}"
-confirm "Install Starship ${GREEN}latest${NO_COLOR} to ${BOLD}${GREEN}${BIN_DIR}${NO_COLOR}?"
-check_bin_dir "${BIN_DIR}"
+if [ -w "$BIN_DIR" ]; then
+  info "Installing Starship, please wait…"
+else
+  warn "Escalated permission are required to install to $BIN_DIR"
+  elevate_priv
+  info "Installing Starship as root, please wait…"
+  elevated=1
+fi
+
+fetch "$archive" "$url"
+unpack "$archive"
+if [ -n "$elevated" ]; then
+  sudo install "$bin" "$BIN_DIR"
+else
+  install "$bin" "$BIN_DIR"
+fi
 
-install
-complete "Starship installed"
+success "Starship installed"
+printf "\n"
```

Comment:
Similar question to before: move this into the success message?

Reply:
The same answer, I would love to but sadly I can't. 😥

Reply:
Oh heck, I just realized....this creates an additional blank line, doesn't it?

I thought it was needed to make the first newline appear. Whoops.

Reply:
Yup, generates additional blank line.